media/CircleEditor/view-sidebar.js

```diff
@@ -1456,30 +1437,250 @@ sidebar.ArgumentView = class {
+                    buf[i] = view.getUint8(0);
+                    break;
+                case 'float64':
+                    view.setFloat64(0, parseFloat(data[i]), true);
+                    for (let j = 0; j < 8; j++) {
+                        buf[i * 8 + j] = view.getUint8(j);
+                    }
+                    break;
+                default:  // TODO Enable other types uint32, uint64, string
+                    break;
+            }
+        }
+
+        const newBuffer = [];
+
+        for (let i = 0; i < Object.keys(original).length; i++) {
+            newBuffer.push(original[i]);
+        }
+    }
+
+    removeBracket(str) {
+        str = str.replace(/\[/g, '');
+        str = str.replace(/\]/g, '');
+        str = str.replace(/\n/g, '');
+        str = str.replace(/ /g, '');
+        str = str.replace(/\{/g, '');
+        str = str.replace(/\}/g, '');
+        str = str.replace(/"/g, '');
+        str = str.replace(/:/g, '');
+        str = str.replace(/low/g, '');
+        str = str.replace(/high/g, '');
+        const arr = str.split(',');
+        return arr;
+    }
+
+    parseBufData(bufData, shape, type) {
+        /* data validation - bracket check */
+        const stack = [];
+        for (let i = 0; i < bufData.length; i++) {
+            if (bufData.charAt(i) === '[') {
+                stack.push('[');
+            } else if (bufData.charAt(i) === ']') {
+                if (stack[stack.length - 1] === '[') {
+                    stack.pop();
+                } else {
+                    // alert(error! Brackets do not match);
+                    return [];
+                }
+            }
+        }
+        if (stack.length) {
+            // alert(error! Brackets do not match);
+            return [];
+        }
+
+
+        /* parse data to string array */
+        var dataArr = this.removeBracket(bufData);
+
+
+        /* data validation - shape count */
+        let shapeCnt = 1;
+        for (let i = 0; i < shape.length; i++) {
+            shapeCnt *= shape[i];
+        }
+        if (type === 'int64') {
+            shapeCnt *= 2;
+        }
+        if (dataArr.length !== shapeCnt) {
+            // alert(error! Shape and data count does not match);
+            return [];
+        }
+
+        return dataArr;
+    }
+
+    /* buffer data modified - type change NOT allowed simultaneously */
+    editBufferData(bufData, currentType, shape) {
+        const dataArr = this.parseBufData(bufData, shape, currentType);
+
+        if (dataArr.length === 0) {
+            vscode.postMessage({command: 'alert', text: 'Validation Error!'});
+            return;
+        }
+
+        /* compare changed elements and update data */
+        if (currentType === 'string') {
+            // TODO string type
+        } else {
+            this.storeData(
+                dataArr, this._editObject._arguments._initializer._data, currentType.toLowerCase());
+        }
```

Comment:
I think `string` is not needed to be treated specially... is it possible just to change as following?

```suggestion
        this.storeData(
            dataArr, this._editObject._arguments._initializer._data, currentType.toLowerCase());
```